src/github.com/mozilla-services/pushgo/simplepush/dynamo_store.go

```diff
@@ -0,0 +1,507 @@
+	success, err = s.table.PutItem(uaid, chid, []dynamodb.Attribute{
+		*dynamodb.NewNumericAttribute(VERS_LABEL, vers),
+		*dynamodb.NewNumericAttribute(MODD_LABEL, now),
+	})
+	if err != nil {
+		return
+	}
+	if success != true {
+		err = ErrDynamoDBFailure
+	}
+	return
+}
+
+// Update updates the version for the given device ID and channel ID.
+// Implements Store.Update().
+func (s *DynamoDBStore) Update(uaid, chid string, version int64) (err error) {
+	now := s.getNow(0)
+	success, err := s.table.UpdateAttributes(&dynamodb.Key{uaid, chid},
+		[]dynamodb.Attribute{
+			*dynamodb.NewNumericAttribute(VERS_LABEL,
+				strconv.FormatInt(version, 10)),
+			*dynamodb.NewNumericAttribute(MODD_LABEL, now),
+		})
+	if err != nil {
+		return
+	}
+	if success != true {
+		err = ErrDynamoDBFailure
+	}
+	return
+}
+
+// Unregister marks the channel ID associated with the given device ID
+// as inactive. Implements Store.Unregister().
+func (s *DynamoDBStore) Unregister(uaid, chid string) (err error) {
+	success, err := s.table.DeleteItem(&dynamodb.Key{uaid, chid})
+	if err != nil {
+		return
+	}
+	if success != true {
+		err = ErrDynamoDBFailure
+	}
+	return
+}
+
+// Drop removes a channel ID associated with the given device ID from
+// memcached. Deregistration calls should call s.Unregister() instead.
+// Implements Store.Drop().
+func (s *DynamoDBStore) Drop(uaid, chid string) (err error) {
+	return s.Unregister(uaid, chid)
+}
+
+// FetchAll returns all channel updates and expired channels for a device ID
+// since the specified cutoff time. Implements Store.FetchAll().
+func (s *DynamoDBStore) FetchAll(uaid string, since time.Time) (updates []Update, expired []string, err error) {
+
+	attrs := []dynamodb.AttributeComparison{
+		*dynamodb.NewEqualStringAttributeComparison(UAID_LABEL, uaid),
+		*dynamodb.NewStringAttributeComparison(CHID_LABEL, dynamodb.COMPARISON_GREATER_THAN, " "),
+	}
+	results, err := s.table.Query(
+		attrs,
+	)
+	if err != nil {
+		return
+	}
+	for _, r := range results {
+		var vers string
+		if vera, ok := r[VERS_LABEL]; ok {
+			vers = vera.Value
+		} else {
+			if vera, ok := r["created"]; ok {
+				vers = vera.Value
+			} else {
+				vers = s.getNow(0)
+			}
+		}
+		version, err := strconv.ParseUint(vers, 10, 64)
+		if err != nil {
+			version = uint64(time.Now().UTC().Unix())
+		}
+		updates = append(updates, Update{
+			ChannelID: r[CHID_LABEL].Value,
+			Version:   version,
+		})
+	}
+
+	return
+}
+
+// DropAll removes all channel records for the given device ID. Implements
+// Store.DropAll().
+func (s *DynamoDBStore) DropAll(uaid string) (err error) {
+	var ok bool
+	result, err := s.table.Query([]dynamodb.AttributeComparison{
+		*dynamodb.NewEqualStringAttributeComparison(UAID_LABEL, uaid),
+	})
+	if err != nil {
+		return
+	}
+	for _, record := range result {
+		key := &dynamodb.Key{
+			HashKey:  record[UAID_LABEL].Value,
+			RangeKey: record[CHID_LABEL].Value,
+		}
+		if ok, err = s.table.DeleteItem(key); !ok {
```

Comment:
This should use the BatchWriteItem should be used with all the DeleteItem's inside it for a single request.